pilot/pkg/bootstrap/server.go

```diff
@@ -335,10 +336,11 @@ func (s *Server) initKubeClient(args *PilotArgs) error {
 			needToCreateClient = true
 		case EurekaRegistry:
 			needToCreateClient = true
-		}
+		case CloudFoundryRegistry:
```

Comment:
if you have multiple registries, you want to create a client if ANY of the registries need it, so we shouldn't explicitly set `needToCreateClient = false` here